src/react/hooks/__tests__/useQuery.test.tsx

```diff
@@ -2092,6 +2092,200 @@ describe("useQuery Hook", () => {
       unmount();
       result.current.stopPolling();
     });
+
+    describe("should stop refetching if `shouldRefetchOnPolling` is passed at", () => {
+      beforeEach(() => {
+        jest.useFakeTimers();
+      });
+
+      afterEach(() => {
+        jest.runOnlyPendingTimers();
+        jest.useRealTimers();
+      });
+
+      it("global level", async () => {
+        document.hasFocus = jest.fn().mockImplementation(() => true);
+
+        const query = gql`
+          {
+            hello
+          }
+        `;
+        const link = mockSingleLink(
+          {
+            request: { query },
+            result: { data: { hello: "world 1" } },
+          },
+          {
+            request: { query },
+            result: { data: { hello: "world 2" } },
+          },
+          {
+            request: { query },
+            result: { data: { hello: "world 3" } },
+          }
+        );
+
+        const client = new ApolloClient({
+          link,
+          cache: new InMemoryCache(),
+          defaultOptions: {
+            watchQuery: {
+              shouldRefetchOnPolling: () => document.hasFocus(),
```

Comment:
```suggestion
              shouldRefetchOnPolling: () => shouldRefetch,
```

From my previous suggestion, let's use a variable here instead to avoid the reference to `document`

Reply:
Defined it as a callback returning boolean value instead of boolean value. Let me know if you want me to change it to boolean only.